src/components/FactoryBlock/BlockConnector.js

```diff
@@ -1,17 +1,28 @@
 import { connect } from 'react-redux'
-import { machineSelected, actionSelected } from '../../selectors/GameState'
+import { machineSelected, actionSelected, moveSelectedNode } from '../../selectors/GameState'
 import { selectMachineNode } from '../../selectors/Grid'
-import { updateBlock, deleteBlock } from '../../actions/Grid'
+import { updateBlock, deleteBlock, moveBlock } from '../../actions/Grid'
+import { selectMoveBlock, deselectMoveBlock } from '../../actions/GameState'
 
 const connector = Block => {
   const mapStateToProps = (state, props) => ({
     machineSelected: machineSelected(state),
     actionSelected: actionSelected(state),
+    moveSelectedNode: moveSelectedNode(state),
     node: selectMachineNode(state, props.position),
   })
   const mapDispatchToProps = dispatch => ({
+    updateBlock: (position, machineType) => {
+      if (machineType) {
+        dispatch(updateBlock(position, machineType))
```

Comment:
Is it working? I think we have to send the entire machine when we update a block.

Reply:
My bad, It works perfectly. I just got confused by changing the `machine` name to `machineType`.
I'm going to aprove this pr.